### PR TITLE
Census migration: Handling tribal consent

### PR DESCRIPTION
**Pull request by phildominguez-gsa**
Closes https://github.com/GSA-TTS/FAC/issues/3033

In this PR:
- Now checking for suppression when the submission is the tribal entity type, and setting the corresponding sac boolean. They are only considered non-public if the entity type is "tribal" and the suppression code is "it".
- The name and title strings are being set as "GSA_MIGRATION" since we don't have them
- Adding unit tests for setting `sac.tribal_data_consent` values

Testing:
- Pick a dbkey and run it through normally. The corresponding entry in the dissemination table should have `is_public` set to true.
- In the `elecauditheader` table , change the `ENTITY_TYPE` to "Tribal" and set `SUPPRESSION_CODE` to "IT" for your dbkey. Run the dbkey again and `is_public` should now be false in the dissemination db.
- For bonus points you can also run a tribal submission with a `SUPPRESSION_CODE` that is **not** "IT" to confirm it makes it public

## PR checklist: submitters

- [ ]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [ ]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [ ]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [ ]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [ ]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [ ]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [ ]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

File: backend/census_historical_migration/sac_general_lib/sac_creator.py [hunk 1]
```diff
@@ -81,6 +81,14 @@ def setup_sac(user, audit_header):
     sac.auditee_certification = auditee_certification(audit_header)
     sac.auditor_certification = auditor_certification(audit_header)
     sac.data_source = settings.CENSUS_DATA_SOURCE
+
+    if audit_header.ENTITY_TYPE == "Tribal":
```
Comment (sambodeme) [inline backend/census_historical_migration/sac_general_lib/sac_creator.py:85]:
```suggestion
    if general_info["entity_type"] == "tribal":

```

File: backend/census_historical_migration/sac_general_lib/sac_creator.py [hunk 2]
```diff
@@ -81,6 +81,14 @@ def setup_sac(user, audit_header):
     sac.auditee_certification = auditee_certification(audit_header)
     sac.auditor_certification = auditor_certification(audit_header)
     sac.data_source = settings.CENSUS_DATA_SOURCE
+
+    if audit_header.ENTITY_TYPE == "Tribal":
+        sac.tribal_data_consent = {
+            "tribal_authorization_certifying_official_title": "GSA_MIGRATION",
+            "is_tribal_information_authorized_to_be_public": audit_header.SUPPRESSION_CODE != "IT",
```
Comment (sambodeme) [inline backend/census_historical_migration/sac_general_lib/sac_creator.py:88]:
```suggestion
            "is_tribal_information_authorized_to_be_public": string_to_string(audit_header.SUPPRESSION_CODE).upper() != "IT",

```

Comment (sambodeme) [reply backend/census_historical_migration/sac_general_lib/sac_creator.py:88]: Are we supposed to make it public when `IT` or when not `IT` ? We may want to document this in the ticket.

File: backend/census_historical_migration/sac_general_lib/sac_creator.py [hunk 3]
```diff
@@ -81,6 +81,14 @@ def setup_sac(user, audit_header):
     sac.auditee_certification = auditee_certification(audit_header)
     sac.auditor_certification = auditor_certification(audit_header)
     sac.data_source = settings.CENSUS_DATA_SOURCE
+
+    if audit_header.ENTITY_TYPE == "Tribal":
+        sac.tribal_data_consent = {
+            "tribal_authorization_certifying_official_title": "GSA_MIGRATION",
```
Comment (sambodeme) [inline backend/census_historical_migration/sac_general_lib/sac_creator.py:87]:
```suggestion
            "tribal_authorization_certifying_official_title": settings.GSA_MIGRATION,

```

File: backend/census_historical_migration/sac_general_lib/sac_creator.py [hunk 4]
```diff
@@ -81,6 +81,14 @@ def setup_sac(user, audit_header):
     sac.auditee_certification = auditee_certification(audit_header)
     sac.auditor_certification = auditor_certification(audit_header)
     sac.data_source = settings.CENSUS_DATA_SOURCE
+
+    if audit_header.ENTITY_TYPE == "Tribal":
+        sac.tribal_data_consent = {
+            "tribal_authorization_certifying_official_title": "GSA_MIGRATION",
+            "is_tribal_information_authorized_to_be_public": audit_header.SUPPRESSION_CODE != "IT",
+            "tribal_authorization_certifying_official_name": "GSA_MIGRATION",
```
Comment (sambodeme) [inline backend/census_historical_migration/sac_general_lib/sac_creator.py:89]:
```suggestion
            "tribal_authorization_certifying_official_name": settings.GSA_MIGRATION,

```

**Review by sambodeme — CHANGES_REQUESTED** [review]

File: backend/census_historical_migration/sac_general_lib/sac_creator.py (hunk 2 above)
Comment (phildominguez-gsa) [reply backend/census_historical_migration/sac_general_lib/sac_creator.py]: My understanding from the meeting with Carley is that if IT is not there, it's public. We can def verify before this goes out. Added that note to the ticket too.

File: backend/census_historical_migration/sac_general_lib/sac_creator.py (hunk 3 above)
Comment (phildominguez-gsa) [reply backend/census_historical_migration/sac_general_lib/sac_creator.py]: Done

File: backend/census_historical_migration/sac_general_lib/sac_creator.py (hunk 4 above)
Comment (phildominguez-gsa) [reply backend/census_historical_migration/sac_general_lib/sac_creator.py]: Done

File: backend/census_historical_migration/sac_general_lib/sac_creator.py (hunk 1 above)
Comment (sambodeme) [reply backend/census_historical_migration/sac_general_lib/sac_creator.py]: That was a typo, `entity_type` is the name in dissemination use `general_info["user_provided_organization_type"]` instead.

File: backend/census_historical_migration/sac_general_lib/sac_creator.py [hunk 5]
```diff
@@ -81,6 +83,15 @@ def setup_sac(user, audit_header):
     sac.auditee_certification = auditee_certification(audit_header)
     sac.auditor_certification = auditor_certification(audit_header)
     sac.data_source = settings.CENSUS_DATA_SOURCE
+
+    if audit_header.ENTITY_TYPE.upper() == "TRIBAL":
```
[the remark this answers is not in the document]
Comment (phildominguez-gsa) [reply backend/census_historical_migration/sac_general_lib/sac_creator.py]: Done

File: backend/census_historical_migration/sac_general_lib/sac_creator.py (hunk 1 above)
Comment (phildominguez-gsa) [reply backend/census_historical_migration/sac_general_lib/sac_creator.py]: Done

**Review by sambodeme — APPROVED** [review]
LGTM.

**Pull request merged**